common/source/world/Biome.hpp

```diff
@@ -0,0 +1,105 @@
+ *
+ *  Copyright (C) 2018-2020 Unarelith, Quentin Bazin <openminer@unarelith.net>
+ *  Copyright (C) 2019-2020 the OpenMiner contributors (see CONTRIBUTORS.md)
+ *
+ *  This file is part of OpenMiner.
+ *
+ *  OpenMiner is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  OpenMiner is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with OpenMiner; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ *
+ * =====================================================================================
+ */
+#ifndef BIOME_HPP_
+#define BIOME_HPP_
+
+#include <string>
+#include <vector>
+
+#include <gk/core/IntTypes.hpp>
+
+#include "FloraPlacementEntry.hpp"
+#include "ISerializable.hpp"
+#include "OrePlacementEntry.hpp"
+#include "TreePlacementEntry.hpp"
+
+class Biome : public ISerializable {
+	public:
+		Biome() = default;
+		Biome(u32 id, const std::string &stringID, const std::string &label);
+
+		void serialize(sf::Packet &packet) const override;
+		void deserialize(sf::Packet &packet) override;
+
+		u16 id() const { return m_id & 0xffff; }
+		const std::string &stringID() const { return m_stringID; }
+		const std::string &label() const { return m_label; }
+		void setLabel(const std::string &label) { m_label = label; }
+
+		const std::vector<double> getParams() const { return m_params; }
+		u16 getTopBlockID() const { return m_topBlockID; }
+		u16 getGroundBlockID() const { return m_groundBlockID; }
+		u16 getDeepBlockID() const { return m_deepBlockID; }
+		u16 getBeachBlockID() const { return m_beachBlockID; }
+		u16 getLiquidBlockID() const { return m_liquidBlockID; }
+		const std::vector<FloraPlacementEntry> &getFlora() const { return m_flora; }
+		const std::vector<OrePlacementEntry> &getOres() const { return m_ores; }
+		const std::vector<TreePlacementEntry> &getTrees() const { return m_trees; }
+
+		void setParams(std::vector<double>& value) { m_params = value; }
+		void setTopBlockID(u16 value) { m_topBlockID = value; }
+		void setGroundBlockID(u16 value) { m_groundBlockID = value; }
+		void setDeepBlockID(u16 value) { m_deepBlockID = value; }
+		void setBeachBlockID(u16 value) { m_beachBlockID = value; }
+		void setLiquidBlockID(u16 value) { m_liquidBlockID = value; }
+		void setFlora(std::vector<FloraPlacementEntry> value) { m_flora = value; }
+		void setOres(std::vector<OrePlacementEntry> value) { m_ores = value; }
+		void setTrees(std::vector<TreePlacementEntry> value) { m_trees = value; }
```

Comment:
`setFlora`, `setOres` and `setTrees` should take a `const std::vector<YourClass> &`